src/main/kotlin/earth/terrarium/techarium/common/blocks/entities/machines/SprinklerBlockEntity.kt

```diff
@@ -0,0 +1,128 @@
+package earth.terrarium.techarium.common.blocks.entities.machines
+
+import com.teamresourceful.resourcefullibkt.common.contains
+import earth.terrarium.techarium.common.blocks.entities.DeployableBlockEntity
+import earth.terrarium.techarium.common.capabilities.blocks.ComponentFluidHandler
+import earth.terrarium.techarium.common.registries.ModAttachments
+import earth.terrarium.techarium.common.registries.ModBlockEntityTypes
+import earth.terrarium.techarium.common.registries.ModBlocks
+import earth.terrarium.techarium.common.registries.ModComponents
+import earth.terrarium.techarium.common.utils.*
+import net.minecraft.core.BlockPos
+import net.minecraft.core.BlockPos.MutableBlockPos
+import net.minecraft.core.component.DataComponentMap
+import net.minecraft.core.particles.ParticleTypes
+import net.minecraft.server.level.ServerLevel
+import net.minecraft.tags.BlockTags
+import net.minecraft.tags.FluidTags
+import net.minecraft.world.level.block.Blocks
+import net.minecraft.world.level.block.BonemealableBlock
+import net.minecraft.world.level.block.LevelEvent
+import net.minecraft.world.level.block.state.BlockState
+import net.minecraft.world.level.block.state.properties.BlockStateProperties.POWERED
+import net.neoforged.neoforge.fluids.FluidType
+import net.neoforged.neoforge.fluids.capability.IFluidHandler
+import software.bernie.geckolib.animation.AnimatableManager
+import software.bernie.geckolib.animation.PlayState
+import software.bernie.geckolib.animation.RawAnimation
+
+
+private const val TICKS_PER_OPERATION = 20 * 2
+private const val DRAIN_AMOUNT = 10
+private val DEPLOY = RawAnimation.begin().thenPlay("deploy").thenLoop("idle")
+private val IDLE = RawAnimation.begin().thenLoop("idle")
+private val WORK = RawAnimation.begin().thenPlay("work")
+
+class SprinklerBlockEntity(
+    pos: BlockPos,
+    state: BlockState
+) : DeployableBlockEntity(ModBlockEntityTypes.sprinkler, pos, state) {
+
+    private var workTicks = 0
+    private val isWorking: Boolean
+        get() = (this.uniqueId + this.ticks) % TICKS_PER_OPERATION == 0L && !this.blockState.getValue(POWERED)
+
+    val tank = ComponentFluidHandler.create(ComponentSlot.INPUT, ModAttachments.singleFluidTank, this) {
+        it.`is`(FluidTags.WATER)
```

Comment:
we don't yet, because thats a FluidStack we only have stuff for vanilla classes rn, will add later but I am gonna resolve this no reason to keep the PR waiting on an rlibkt update